plugins/pulp_python/plugins/importers/sync.py

```diff
@@ -39,117 +39,103 @@ def __init__(self, *args, **kwargs):
-        to free memory.
+        This method is bound to a Nectar event listener which is called when the step fails to
+        download json metadata for a particular project. It closes the StringIO that we were using
+        to store the download to free memory.
 
         :param report: The report that details the download
         :type  report: nectar.report.DownloadReport
         """
         report.destination.close()
-
         super(DownloadMetadataStep, self).download_failed(report)
 
     def download_succeeded(self, report):
         """
-        This method is called by Nectar for each package metadata file after it is successfully
-        downloaded. It reads the manifest and adds each available unit to the parent step's
-        "available_units" attribute.
-
-        It also adds each unit's download URL to the parent step's "unit_urls" attribute. Each key
-        is a models.Package object.
+        This method is bound to a Nectar event listener which is called each time the step
+        successfully downloads json metadata for a project. Each package described in the project's
+        json is initialized and placed in the parent step's list of units that are available to be
+        downloaded.
 
         :param report: The report that details the download
         :type  report: nectar.report.DownloadReport
         """
         _logger.info(_('Processing metadata retrieved from %(url)s.') % {'url': report.url})
         with contextlib.closing(report.destination) as destination:
             destination.seek(0)
-            self._process_manifest(destination.read())
+            self._process_metadata(destination.read())
 
         super(DownloadMetadataStep, self).download_succeeded(report)
 
     def generate_download_requests(self):
         """
-        For each package name that the user has requested, yield a Nectar DownloadRequest for its
-        metadata file in JSON format.
+        For each project name in the parent step's list of projects, yield a Nectar DownloadRequest
+        for the project's json metadata.
 
-        :return: A generator that yields DownloadRequests for the metadata files.
+        :return: A generator that yields DownloadRequests for a project's json metadata.
         :rtype:  generator
         """
-        # We need to retrieve the manifests for each of our packages
-        manifest_urls = [urljoin(self.parent._feed_url, 'pypi/%s/json/' % pn)
-                         for pn in self.parent._package_names]
-        for u in manifest_urls:
+        metadata_urls = [urljoin(self.parent._feed_url, 'pypi/%s/json/' % pn)
+                         for pn in self.parent._project_names]
+        for u in metadata_urls:
             if self.canceled:
                 return
             yield request.DownloadRequest(u, StringIO(), {})
 
-    def _process_manifest(self, manifest):
+    def _process_metadata(self, metadata):
```

Comment:
Ah. That is unfortunate. How do you feel about a ``Project`` model, then? It sounded like you were considering it, and this could live on that model for now.